crawl-ref/source/travel.cc

```diff
@@ -3962,8 +3962,13 @@ void TravelCache::add_waypoint(int x, int y)
         return;
     }
 
-    int waynum = keyin - '0';
+    set_waypoint(keyin - '0', x, y);
 
+}
+
+void TravelCache::set_waypoint(int waynum, int x, int y)
+{
+    // Assumes that the waynum input is a valid number between 0 and 9 inclusive.
```

Comment:
Might be good to instead of this comment to add an ASSERT, e.g. `ASSERT_RANGE(waynum, 0, 10);`. In this case, there's also a #define for the upper bound (in travel.h), so even better would be: `ASSERT_RANGE(waynum, 0, TRAVEL_WAYPOINT_COUNT);`

Reply:
done